vllm/entrypoints/openai/serving_chat.py

```diff
@@ -222,7 +222,22 @@ async def chat_completion_stream_generator(
                             model=model_name)
                         if (request.stream_options
                                 and request.stream_options.include_usage):
-                            chunk.usage = None
+                            if (request.stream_options.continuous_usage_stats
+                                    or res.outputs[i].finish_reason
+                                    is not None):
+                                prompt_tokens = len(res.prompt_token_ids)
+                                completion_tokens = 0
+                                usage = UsageInfo(
+                                    prompt_tokens=prompt_tokens,
+                                    completion_tokens=completion_tokens,
+                                    total_tokens=prompt_tokens +
+                                    completion_tokens,
+                                )
+                            if request.stream_options.continuous_usage_stats:
+                                chunk.usage = usage
+                            else:
+                                chunk.usage = None
```

Comment:
Perhaps this could just be simplified to:
```python
if (request.stream_options
        and request.stream_options.include_usage):
    if request.stream_options.continuous_usage_stats:
        prompt_tokens=len(res.prompt_token_ids)
        usage = UsageInfo(
            prompt_tokens=prompt_tokens,
            completion_tokens=0,
            total_tokens=prompt_tokens,
        )
    else:
        chunk=None
```